pybamm/expression_tree/printing/sympy_overrides.py

```diff
@@ -3,22 +3,27 @@
 #
 import re
 
-from sympy.printing.latex import LatexPrinter
+from pybamm.util import have_optional_dependency
 
 
-class CustomPrint(LatexPrinter):
+def custom_latex_printer(expr, **settings):
+    latex = have_optional_dependency("sympy", "latex")
+    Derivative = have_optional_dependency("sympy", "Derivative")
+    if isinstance(expr, Derivative) and getattr(expr, "force_partial", False):
+        latex_str = latex(expr, **settings)
+        var1, var2 = re.findall(r"^\\frac{(\w+)}{(\w+) .+", latex_str)[0]
+        latex_str = latex_str.replace(var1, "\partial").replace(var2, "\partial")
+        return latex_str
+    else:
+        return latex(expr, **settings)
+
+class CustomPrint:
     """Override SymPy methods to match PyBaMM's requirements"""
 
     def _print_Derivative(self, expr):
         """Override :meth:`sympy.printing.latex.LatexPrinter._print_Derivative`"""
-        eqn = super()._print_Derivative(expr)
-
-        if getattr(expr, "force_partial", False) and "partial" not in eqn:
-            var1, var2 = re.findall(r"^\\frac{(\w+)}{(\w+) .+", eqn)[0]
-            eqn = eqn.replace(var1, "\partial").replace(var2, "\partial")
-
-        return eqn
-
+        return custom_latex_printer(expr)
```

Comment:
I am not sure if this will work or if this is already working (cc: @priyanshuone6 @brosaplanella).

If this is in working condition  - we should get rid of the `CustomPrint` and `custom_print_func`. Both of them are redundant because everything should work if we pass `custom_latex_printer` directly here (the only place `custom_print_func` is used) -
https://github.com/pybamm-team/PyBaMM/blob/b0db430a65cb0c6a964f139c92a6e1346d8dc36a/pybamm/expression_tree/operations/latexify.py#L227-L233

Reply:
I am not very familiar with this part of the code, @priyanshuone6 will know better. It seems to be related to printing partial derivatives, so I would say that as long as partial derivatives get printed fine  (you can check with SPM or DFN), then I am happy with whatever implementation makes our lives easier.

Reply:
The current fix does not work :(

<img width="461" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/74055102/89190e13-0726-4549-bd9a-0a16d02950cc">

I think this should be a partial derivative in SPM?

Reply:
Yes, this should be partial

Reply:
Fixed it (reverted back to the original class)!